src/DataStructures.jl

```diff
@@ -14,11 +14,11 @@ module DataStructures
                  ReverseOrdering, Reverse, Lt,
                  isless,
                  union, intersect, symdiff, setdiff, issubset,
-                 find, searchsortedfirst, searchsortedlast, endof, in
+                 find, searchsortedfirst, searchsortedlast, endof, in, produce
 
 
     export Deque, Stack, Queue
-    export deque, enqueue!, dequeue!, update!,iter
+    export deque, enqueue!, dequeue!, update!, forward_iter, backward_iter
```

Comment:
@lindahua, who introduced those names, is coming from a C++ background, and in C++, the corresponding iterators are `forward_iterator` and `backward_iterator`. 

Thinking more, I would actually prefer that, instead, we had a natural order iterator (the default), and a second, `reverse_iter`, which is just opposite of the default iteration.  Reasons:

1. It's easier to reason about--you don't have to understand how the underlying data structure is implemented, you only have to understand its default iteration order.
2. At one point, a generic `reverse` iterator (and corresponding `Reverse` type) was implemented this way in Julia (it was later changed to return a concrete value).
3. It's one less function to create.